core/meta/src/TClass.cxx

```diff
@@ -824,6 +824,8 @@ void TBuildRealData::Inspect(TClass* cl, const char* pname, const char* mname, c
       TRealData::GetName(rdName,dm);
       rname += rdName;
       TRealData* rd = new TRealData(rname.Data(), offset, dm);
+      if (isTransientMember || IsNestedTransient())
```

Comment:
Could you clarify why this code has been put here and was not there before?

Reply:
This explained a bit in the commit log:  [`322955a` (#16202)](https://github.com/root-project/root/pull/16202/commits/322955af2e3b37b5bcbaf349e3c581407440c3b1)
Do we need more details?